src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/LeaseAcquirer.java

```diff
@@ -26,12 +26,26 @@ public LeaseAcquirer(LeaseClientProvider leaseClientProvider) {
     }
 
     /**
-     * Main wrapper for blobs to be leased by {@link BlobLeaseClient}.
+     * Main wrapper for blobs to be leased by {@link BlobLeaseClient} and perform non-delete task.
+     * @param blobClient Represents blob
+     * @param onSuccess Runnable task to perform when lease is acquired
+     * @param onFailure Extra step to execute in case an error occurred
+     */
+    public void processAndRelease(BlobClient blobClient, Runnable onSuccess, Consumer<BlobErrorCode> onFailure) {
+        var leaseClient = ifAcquiredOrElse(blobClient, leaseId -> onSuccess.run(), onFailure);
+
+        if (leaseClient != null) {
+            release(leaseClient, blobClient);
+        }
+    }
+
+    /**
+     * Main wrapper for blobs to be leased by {@link BlobLeaseClient} and perform deletion task.
      * @param blobClient Represents blob
      * @param onSuccess Consumer which takes in {@code leaseId} acquired with {@link BlobLeaseClient}
      * @param onFailure Extra step to execute in case an error occurred
      */
-    public void ifAcquiredOrElse(
+    public BlobLeaseClient ifAcquiredOrElse(
```

Comment:
How about simply adding a boolean parameter to this method?

Then no `processAndRelease` method (that does not even actually always release) will be needed and this class will not be concerned what kind of task is being done (new javadoc "and perform non-delete task" / "and perform non-delete task."

Reply:
ok